### ISSUE TITLE
Calling `enable` creates improper View data if player was playing while disabled

### ISSUE THREAD

**Issue by cole-smith** [muxinc/mux-stats-sdk-media3#88]
Apologies for the title; summarizing this is hard 😆 

As discussed in [a previous issue](https://github.com/muxinc/mux-stats-sdk-media3/issues/84), we use a single ExoPlayer for all of our playback, and we'd like to disable Mux stats collection for some content and not others. With that in mind, use case would look like this:
```kotlin
// play non-Mux content; disable mux stats during playback
muxStats.disable()
isMuxStatsDisabled = true
exoPlayer.setCurrentItem(nonMuxItem)
exoPlayer.prepare()
exoPlayer.play()
...
// play Mux content; re-enable mux stats during playback
exoPlayer.stop()
exoPlayer.setCurrentItem(muxItem)
if (isMuxStatsDisabled) {
    muxStats.enable(muxItemCustomerData)
} else {
    muxStats.onVideoChange(muxItemCustomerData.videoData)
}
exoPlayer.prepare()
exoPlayer.play()
```

When switching from "disabled" to "enabled" content, it seems that calling `enable` after other content was already playing leads to glitched View data where there's no "Starting Up" time.
<img width="1194" alt="Screenshot 2024-08-05 at 12 58 08 PM" src="https://github.com/user-attachments/assets/246c938d-d712-4550-b242-f5f6c21a107a">
The "paused" region in the screenshot above should reflect the exoPlayer's "starting up" time where it's preparing the new media source. You can also see from the "Playhead time" that the first 5 events correspond to the previous "disabled" media item and not the new item.

If switching between content without needing to "enable" the muxStats, however, `onVideoChanged` handles this properly properly, as seen in the screenshot below.
<img width="1190" alt="Screenshot 2024-08-05 at 1 26 39 PM" src="https://github.com/user-attachments/assets/5be0a5d2-26f3-4549-bde5-72ab7d2c95c9">

Some fixes I've tried, to no avail:
- Calling `onVideoChange` before `enable`, calling `onVideoChange` after `enable`
- Calling (and not calling) `exoPlayer.stop()` before setting the new item and calling `enable`
  - If I remove the call to `exoPlayer.stop()` prior to `exoPlayer.setCurrentItem(muxItem)`, then the glitched "pause" region is shown as "rebuffering" instead
- Calling `enable` before calling `exoPlayer.setCurrentItem` and `exoPlayer.prepare()`

Would it be possible for `enable` to cover the same case that `onVideoChanged` covers? Otherwise, it only works effectively if the player is idle while mux stats is "disabled" prior to re-enabling it.

**Comment by daytime-em** [comment]
Thank you for your detailed report. 

"Starting up" is the intended state for this case. Sorry for exposing this bug, I should be able to find a fix pretty soon.

**Comment by cole-smith** [comment]
Oops, didn't mean to do that lol. Ty for looking into it!!

**Comment by cole-smith** [comment]
@daytime-em I might've been mistaken about `onVideoChange` always handling the above case perfectly as well. I've been able find some Views that have missing "Starting Up" blocks prior to us using the `enable/disable` functionality. Can hop on a call or something to explain further if you can't repro on your end

**Comment by daytime-em** [comment]
@cole-smith I was able to reproduce the bugs you mentioned in your report, including some issues with startup-time after `videoChange`. 

The fix should be available soon. If you want to verify it before we release, you can clone [this branch](https://github.com/muxinc/mux-stats-sdk-media3/tree/fix/enable-catchup-logic) and run the sample app. It just needs an `ANDROID_HOME` or `sdk.dir` to build. You can also use `publishToMavenLocal` to test the library in your own app. The version name of the build is printed in the build output, something like `dev-fix-enable-catchup-logic-0c3aabe`.

If you'd like to schedule a call with me about any of this, I think the best way would be to go through Glenn.

**Comment by cole-smith** [comment]
@daytime-em Amazing! Thanks for the quick turnaround :) I'll try to test it out if I get time, but I might not be able to get to it this week. Appreciate the help!
